### PR TITLE
[WIP] Added PlayerFlowerPotManipulateEvent

### PR DESCRIPTION
**Pull request by MisterVector**
Added an event for when players use the "use" button to manipulate the contents of flower pots.

Unfortunately, the server-side code that handles cancellation doesn't properly tell the client the state of the flower pot, so if the event was cancelled, the client thinks an item was removed, or the item was placed into the pot. I tried tileentityflowerpot.update() and a few other things to no avail. If I had some pointers that'd be awesome!

### CODE REVIEW

**Comment by Black-Hole** [comment]
A new event just for this? How about sending the original block back to player if the interact is cancelled?

File: Spigot-API-Patches/0042-Added-PlayerFlowerPotManipulateEvent-for-manipulatin.patch [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
++import javax.annotation.Nonnull;
++import org.bukkit.block.Block;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.ItemStack;
++
++/**
++ * Called when a player places or takes an item in a flowerpot
```
Comment (willies952002) [inline Spigot-API-Patches/0042-Added-PlayerFlowerPotManipulateEvent-for-manipulatin.patch:25]: Suggestion: `Called when a player places a plant in a flowerpot`

Comment (MisterVector) [reply Spigot-API-Patches/0042-Added-PlayerFlowerPotManipulateEvent-for-manipulatin.patch:25]: But what if they're taking the plant out of the flowerpot?

Comment (phase) [reply Spigot-API-Patches/0042-Added-PlayerFlowerPotManipulateEvent-for-manipulatin.patch:25]: `Called when a player places a plant in or takes a plant out of a flowerpot.`

Comment (willies952002) [reply Spigot-API-Patches/0042-Added-PlayerFlowerPotManipulateEvent-for-manipulatin.patch:25]: ~~Last I checked, the only way to take a plant out of a flower pot is to actually break the pot..~~

I stand corrected:
"The plant can be removed from the flower pot by pressing the use control." - MC Wiki

**Comment by kashike** [comment]
x

**Comment by MisterVector** [comment]
I'm going to remake this branch and then PR on top of the latest paper changes.

**Pull request closed without merging**